administrator/components/com_jedchecker/libraries/rules/language.php

```diff
@@ -0,0 +1,177 @@
+<?php
+/**
+ * @package    Joomla.JEDChecker
+ *
+ * @copyright  Copyright (C) 2021 Open Source Matters, Inc. All rights reserved.
+ *
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die('Restricted access');
+
+
+// Include the rule base class
+require_once JPATH_COMPONENT_ADMINISTRATOR . '/models/rule.php';
+
+
+/**
+ * class JedcheckerRulesLanguage
+ *
+ * This class validates language ini file
+ *
+ * @since  2.3
+ */
+class JedcheckerRulesLanguage extends JEDcheckerRule
+{
+	/**
+	 * The formal ID of this rule. For example: SE1.
+	 *
+	 * @var    string
+	 */
+	protected $id = 'LANG';
+
+	/**
+	 * The title or caption of this rule.
+	 *
+	 * @var    string
+	 */
+	protected $title = 'COM_JEDCHECKER_LANG';
+
+	/**
+	 * The description of this rule.
+	 *
+	 * @var    string
+	 */
+	protected $description = 'COM_JEDCHECKER_LANG_DESC';
+
+	/**
+	 * Initiates the search and check
+	 *
+	 * @return    void
+	 */
+	public function check()
+	{
+		// Find all INI files of the extension (in the format tag.extension.ini or tag.extension.sys.ini)
+		$files = JFolder::files($this->basedir, '^[a-z]{2,3}-[A-Z]{2}\.\w+(?:\.sys)?\.ini$', true, true);
+
+		// Iterate through all the ini files
+		foreach ($files as $file)
+		{
+			// Try to validate the file
+			$this->find($file);
+		}
+	}
+
+	/**
+	 * Reads and validates an ini file
+	 *
+	 * @param   string  $file  - The path to the file
+	 *
+	 * @return bool True on success, otherwise False.
+	 */
+	protected function find($file)
+	{
+		$lines = file($file);
+		foreach ($lines as $lineno => $line)
+		{
+			$line = trim($line);
+
+			// Check for BOM sequence
+			if ($lineno === 0 && strncmp($line, "\xEF\xBB\xBF", 3) === 0)
+			{
+				$this->report->addError($file, JText::_('COM_JEDCHECKER_LANG_BOM_FOUND'), 1);
```

Comment:
Please make this a warning, not an error. The file will still work, but its not advisable...

Reply:
In j3-dev branch I've finally transformed it into the following check: https://github.com/dryabov/jedchecker/commit/beef10df1ee85f865ebeaac7c5130975adc79fd9
I.e. display error in the case of BOM followed by a key (because BOM become a part of the key, and the first line translation is "missed"), and warning in the case of first line is a comment (quite a common case). If you agree, I'll pick that commit to this PR.